sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -500,11 +505,22 @@ private void ensureWindowsAreASet() {
     return FullWindowedValueCoder.of(valueCoder, windowCoder);
   }
 
-  /** Returns the {@code ValueOnlyCoder} from the given valueCoder. */
+  /**
+   * Returns the {@code ValueOnlyCoder} from the given valueCoder.
+   *
+   * @deprecated Use {@link #getParamWindowedValueCoder(Coder)} instead. The {@link
+   *     ValueOnlyWindowedValueCoder} is deprecated now.
+   */
+  @Deprecated
   public static <T> ValueOnlyWindowedValueCoder<T> getValueOnlyCoder(Coder<T> valueCoder) {
```

Comment:
I'm not sure we need to deprecate this method and replace all of the invocations. This is still a useful method if no parameters are supplied. We can simply redirect to the new ParamWindowedValueCoder here.

Reply:
Hi, return `ParamWindowedValueCoder` would result some compatibility problems for Dataflow, but I think we can keep the `getValueOnlyCoder` as you suggested. We can deprecate it later once we find need to.:)